infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/PrepareStorage.java

```diff
@@ -29,26 +32,41 @@
 @Beta
 public class PrepareStorage extends BrokerPhase {
 
+  private static final String SPAN_NAME = "PrepareStorage";
+
   private final String workspaceId;
   private final KubernetesEnvironment brokerEnvironment;
   private final WorkspaceVolumesStrategy volumesStrategy;
   private final StartSynchronizer startSynchronizer;
+  private final Tracer tracer;
 
   public PrepareStorage(
       String workspaceId,
       KubernetesEnvironment brokerEnvironment,
       WorkspaceVolumesStrategy volumesStrategy,
-      StartSynchronizer startSynchronizer) {
+      StartSynchronizer startSynchronizer,
+      Tracer tracer) {
     this.workspaceId = workspaceId;
     this.brokerEnvironment = brokerEnvironment;
     this.volumesStrategy = volumesStrategy;
     this.startSynchronizer = startSynchronizer;
+    this.tracer = tracer;
   }
 
   @Override
   public List<ChePlugin> execute() throws InfrastructureException {
-    volumesStrategy.prepare(
-        brokerEnvironment, workspaceId, startSynchronizer.getStartTimeoutMillis());
+    Span tracingSpan = tracer.buildSpan(SPAN_NAME).start();
+    TracingUtil.setWorkspaceId(tracingSpan, workspaceId);
+
+    try {
+      volumesStrategy.prepare(
+          brokerEnvironment, workspaceId, startSynchronizer.getStartTimeoutMillis());
+    } catch (InfrastructureException e) {
+      TracingUtil.setErrorStatus(tracingSpan, e);
+      tracingSpan.finish();
+      throw e;
+    }
```

Comment:
Can we use finally here? It can help if unchecked exception is thrown for some reason.

Reply:
Good point. Done.

WDYT about similar for `DeployBroker`? Can `nextPhase.execute() be moved out of the `try` block?

Reply:
Finally is also repoving brokers pod, so no - I don't think we can move it out of the finally block

Reply:
:+1: